tests/test_demand_data.cpp

```diff
@@ -35,12 +35,14 @@ boost::test_tools::predicate_result check_string(std::string test, std::string r
 
 BOOST_AUTO_TEST_SUITE(test_demand_data)
 
-
+////  The re-factored version of demand.c is no longer compatible  ////
```

Comment:
One of the points of my PR was to show how true unit testing can be accomplished in the existing code base. I would prefer if you would write tests for your implementation, rather than commenting out tests. Not surprisingly, test coverage went down for this PR.